operator/internal/controller/inflight.go

```diff
@@ -0,0 +1,66 @@
+package controller
+
+import (
+	"sync"
+	"time"
+)
+
+// InFlight is a struct used to manage in flight requests for a unique identifier.
+type InFlight struct {
+	mux                *sync.RWMutex
+	inFlight           map[string]time.Time
```

Comment:
I highly recommend using the [maps](https://github.com/goradd/maps) package (`maps.SafeMap`), for asynchronous map operations

Reply:
Going to hold off on this for now just to avoid extra deps and bc this concurrent map is simple, but will keep that in mind for the future 🙇🏻